io/zenoh-links/zenoh-link-vsock/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+//! ⚠️ WARNING ⚠️
+//!
+//! This crate is intended for Zenoh's internal use.
+//!
+//! [Click here for Zenoh's documentation](../zenoh/index.html)
+use async_trait::async_trait;
+use tokio_vsock::{
+    VsockAddr, VMADDR_CID_ANY, VMADDR_CID_HOST, VMADDR_CID_HYPERVISOR, VMADDR_CID_LOCAL,
+};
+use zenoh_core::zconfigurable;
+use zenoh_link_commons::LocatorInspector;
+use zenoh_protocol::core::{endpoint::Address, Locator};
+use zenoh_result::{zerror, ZResult};
+
+mod unicast;
+pub use unicast::*;
+
+pub const VSOCK_LOCATOR_PREFIX: &str = "vsock";
+
+pub const VSOCK_ADDR_VMADDR_CID_ANY: &str = "VMADDR_CID_ANY";
+pub const VSOCK_ADDR_VMADDR_CID_HYPERVISOR: &str = "VMADDR_CID_HYPERVISOR";
+pub const VSOCK_ADDR_VMADDR_CID_LOCAL: &str = "VMADDR_CID_LOCAL";
+pub const VSOCK_ADDR_VMADDR_CID_HOST: &str = "VMADDR_CID_HOST";
+
+#[derive(Default, Clone, Copy)]
+pub struct VsockLocatorInspector;
+#[async_trait]
+impl LocatorInspector for VsockLocatorInspector {
+    fn protocol(&self) -> &str {
+        VSOCK_LOCATOR_PREFIX
+    }
+
+    async fn is_multicast(&self, _locator: &Locator) -> ZResult<bool> {
+        Ok(false)
+    }
+}
+
+zconfigurable! {
+    // Default MTU in bytes.
+    static ref VSOCK_DEFAULT_MTU: u16 = u16::MAX;
+    // Amount of time in microseconds to throttle the accept loop upon an error.
+    // Default set to 100 ms.
+    static ref TCP_ACCEPT_THROTTLE_TIME: u64 = 100_000;
+}
+
+pub fn get_vsock_addr(address: Address<'_>) -> ZResult<VsockAddr> {
+    let parts: Vec<&str> = address.as_str().split(':').collect();
+
+    if parts.len() != 2 {
+        zerror!("Incorrect vsock address: {:?}", address);
```

Comment:
`zerror!(...)` should be used to create an error to be returned. In this case, you could use the `bail!(...)` macro to create a `ZError` and return.